### PR TITLE
Enable peakviewer for NonOrthogonal view in sliceviewer

### LINKED ISSUE #18299: Re enable peaks viewer for nonOrthogonal view in sliceviewer
Currently peaksviewer is disabled when non orthogonal view in the slice viewer is enabled. 

Blocked until #12110 is merged

### PR DESCRIPTION
**Pull request by lottiegreenwood**
~~**Requires https://github.com/mantidproject/mantid/pull/18958 to be merged first**~~
Fixes #18299 

 - To test go to `\olympic\Babylon5\Scratch\Sam\md.nsx`
 - load `md.nxs` in the sliceviewer, and test with the peak file `37828_peak_table_nuclear.nxs.`
 - See that the skewed peak view looks sensible.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/SliceViewer/src/CompositePeaksPresenter.cpp [hunk 1]
```diff
@@ -55,17 +55,27 @@ void CompositePeaksPresenter::updateWithSlicePoint(
...
 }
 
+void CompositePeaksPresenter::setNonOrthogonal(bool nonOrthogonalEnabled) {
+  if (useDefault()) {
+    m_default->setNonOrthogonal(nonOrthogonalEnabled);
+  }
+  bool result = true;
+  for (auto it = m_subjects.begin(); it != m_subjects.end(); ++it) {
+    (*it)->setNonOrthogonal(nonOrthogonalEnabled);
+  }
```
Comment (AntonPiccardoSelg) [inline MantidQt/SliceViewer/src/CompositePeaksPresenter.cpp]: Avoid the iterator and go directly to the range-based for loop

**Review by AntonPiccardoSelg — CHANGES_REQUESTED** [review]

**Comment by martyngigg** [comment]
@NickDraper Who can work on this now?

**Comment by NickDraper** [comment]
@Mantid-Matthew Could you take on the developer role for this PR please

**Comment by ElliotAOram** [comment]
@samueljackson92 is this ready to test and if so have you got the file around?

**Comment by samueljackson92** [comment]
I don't have that file, but I can probably find you *a* file somewhere

**Comment by martyngigg** [comment]
@samueljackson92 As this is quite old I'm going to rebase the branch with master and get some fresh builds on it.

**Review by martyngigg — APPROVED** [review]
This seems to work as expected. The peaks are plotted in the correct places in the non-orthogonal view.

**Comment by NickDraper** [comment]
We seem to be short of release notes here

**Comment by NickDraper** [comment]
A nice screenshot would not go amiss either

**Pull request merged**